src/main/java/bio/terra/service/dataset/flight/ingest/SkippableStep.java

```diff
@@ -0,0 +1,49 @@
+package bio.terra.service.dataset.flight.ingest;
+
+import bio.terra.stairway.FlightContext;
+import bio.terra.stairway.Step;
+import bio.terra.stairway.StepResult;
+import bio.terra.stairway.exception.RetryException;
+import java.util.function.Predicate;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public abstract class SkippableStep implements Step {
+
+  private final Logger logger = LoggerFactory.getLogger(this.getClass());
+
+  private final Predicate<FlightContext> skipCondition;
+
+  public SkippableStep() {
+    this.skipCondition = x -> false;
```

Comment:
When possible I like chained constructors, even in trivial cases like this. The usual rule is to chain from fewer arguments to more arguments.
```suggestion
   this(x -> false);
```